pkg/registration/hub/managedcluster/controller.go

```diff
@@ -103,6 +110,15 @@ func (c *managedClusterController) sync(ctx context.Context, syncCtx factory.Syn
 	}
 
 	if !managedCluster.Spec.HubAcceptsClient {
+		// If the ManagedClusterAutoApproval feature is enabled, we automatically accept this cluster and
+		// add an annotation to mark its initial join as accepted.
```

Comment:
The comment does is not clear about the purpose of the annotation.

Based on the new test, we want to approve the cluster once on the first time we discover it, so the cluster admin can deny the cluster. This should be explained in the comment.

Reply:
I will update this make it more clear